app/src/main/java/com/mapbox/maps/testapp/examples/viewport/ViewportShowcaseActivity.kt

```diff
@@ -0,0 +1,237 @@
+package com.mapbox.maps.testapp.examples.viewport
+
+import android.annotation.SuppressLint
+import android.content.res.Configuration
+import android.content.res.Resources
+import android.location.Location
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.interpolator.view.animation.FastOutSlowInInterpolator
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.MapView
+import com.mapbox.maps.MapboxMap
+import com.mapbox.maps.plugin.LocationPuck2D
+import com.mapbox.maps.plugin.locationcomponent.LocationConsumer
+import com.mapbox.maps.plugin.locationcomponent.LocationProvider
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.plugin.viewport.ViewportPlugin
+import com.mapbox.maps.plugin.viewport.data.MapboxViewportDataSource
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState
+import com.mapbox.maps.plugin.viewport.viewport
+import com.mapbox.maps.testapp.R
+import com.mapbox.maps.testapp.databinding.ActivityViewportAnimationBinding
+
+class ViewportShowcaseActivity : AppCompatActivity() {
+  private var emitCount = 0
+  private var delta = 0f
+  private val handler = Handler(Looper.getMainLooper())
+  private lateinit var mapboxMap: MapboxMap
+  private lateinit var mapView: MapView
+  private lateinit var viewportButton: Button
+  private val pixelDensity = Resources.getSystem().displayMetrics.density
```

Comment:
we avoid getting density thru singleton as this does not work for multi-screen use-case

Reply:
We need to revisit this later, as these logic should move to the plugin. And we don't have a valid context at this phase, using the pixelRatio from the context would require bigger refactor.